src/main/java/org/apache/maven/plugins/dependency/tree/XMLDependencyNodeVisitor.java

```diff
@@ -0,0 +1,187 @@
+    implements DependencyNodeVisitor
+{
+    /**
+     * Constructor.
+     *
+     * @param writer the writer to write to.
+     */
+    public XMLDependencyNodeVisitor( Writer writer )
+    {
+        super( writer );
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean visit( DependencyNode node )
+    {
+        try
+        {
+            if ( node.getParent() == null || node.getParent() == node )
+            {
+                DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+                DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+                Document doc = dBuilder.newDocument();
+                Element rootElement = getNode( doc, node, true );
+                doc.appendChild( rootElement );
+
+                List<DependencyNode> children = node.getChildren();
+                for ( DependencyNode child : children )
+                {
+                    handleChild( doc, child, rootElement );
+                }
+
+                TransformerFactory transformerFactory = TransformerFactory.newInstance();
+                Transformer transformer = transformerFactory.newTransformer();
+                transformer.setOutputProperty( OutputKeys.INDENT, "yes" );
+                transformer.setOutputProperty( "{http://xml.apache.org/xslt}indent-amount", "2" );
+                DOMSource source = new DOMSource( doc );
+
+                StreamResult console = new StreamResult( writer );
+
+                transformer.transform( source, console );
+            }
+        }
+        catch ( ParserConfigurationException | TransformerException e )
+        {
+            LoggerFactory.getLogger(XMLDependencyNodeVisitor.class).error(e.getMessage());
+        }
+
+        return true;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean endVisit( DependencyNode node )
+    {
+        return true;
+    }
+
+    /**
+     * Render child with its children recursively
+     *
+     * @param doc Docuemnt to use
+     * @param node child node to handle
+     * @param depth depth of the child
+     */
+    private void handleChild( Document doc, DependencyNode node, Element parent )
+    {
+        Element element = getNode( doc, node, false );
+        Node dependencies = parent.getElementsByTagName( "dependencies" ).item( 0 );
+        dependencies.appendChild( element );
+
+        List<DependencyNode> children = node.getChildren();
+        for ( DependencyNode child : children )
+        {
+            handleChild( doc, child, element );
+        }
+    }
+
+    /**
+     * Get element from node
+     *
+     * @param doc Docuemnt to use
+     * @param node Node to get data from
+     */
+    private Element getNode( Document doc, DependencyNode node, boolean root )
+    {
+        Artifact artifact = node.getArtifact();
+        Element element = null;
+
+        if ( root )
+        {
+            element = doc.createElement( "project" );
```

Comment:
A separate method for the root rather than an if statement would reduce cyclomatic complexity